libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3820,6 +3919,148 @@ void TypeChecker::endVisit(UsingForDirective const& _usingFor)
 					": " +  result.message()
 				)
 			);
+		else if (operator_)
+		{
+			if (!_usingFor.typeName()->annotation().type->typeDefinition())
+			{
+				m_errorReporter.typeError(
+					5332_error,
+					path->location(),
+					"Operators can only be implemented for user-defined types and not for contracts."
```

Comment:
It does not sound very natural. Also, does not mention enums, which are disallowed too. Maybe this would be better:
```suggestion
					"Operators can only be implemented for user-defined types that are not contracts or enums."
```

Still not great so maybe better to just negate it?
```suggestion
					"Operators can only be implemented for user-defined value types and structs."
```

Reply:
Yes, now it definitely  sounds better.